lib/Sema/CSSimplify.cpp

```diff
@@ -7993,6 +8012,15 @@ ConstraintSystem::simplifyOptionalObjectConstraint(
       }
     }
 
+    // Provides us with Optional Eval
```

Comment:
Nit: Seems like this comment should be expanded to talk about reasons why that's done.